app/models/organization_table_info.rb

```diff
@@ -0,0 +1,165 @@
+      .select('
+        organizations.id, organizations.api_id, organizations.name, organizations.display_name,
+        SUM(test_payloads.results_count) AS test_results_count, COUNT(test_payloads.id) as test_payloads_count
+      ')
+      .joins('
+        LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+        LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+        LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+      ')
+      .group('organizations.id')
+      .order('test_results_count DESC NULLS LAST')
+
+    # Retrieve the results trends for organizations
+    org_results_trends_count = Organization
+      .select("
+        organizations.id,
+        SUM(test_payloads.results_count) AS test_results_count,
+        date_trunc('week', test_payloads.processed_at) AS trend_date
+      ")
+      .joins('
+        LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+        LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+        LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+      ')
+      .group("organizations.id, date_trunc('week', test_payloads.processed_at)")
+      .order('organizations.id, trend_date ASC')
+      .where("date_trunc('week', test_payloads.processed_at) >= ?", weeks.weeks.ago.at_beginning_of_week)
+
+    # Add where clause to scope to specific organization
+    if organization
+      org_payloads_and_results_count = org_payloads_and_results_count.where('organizations.id = ?', organization.id)
+      org_results_trends_count = org_results_trends_count.where('organizations.id = ?', organization.id)
+    end
+
+    # Add a limit to do the top n
+    if top > 0
+      org_payloads_and_results_count = org_payloads_and_results_count.limit(top)
+    end
+
+    # Collect the ids
+    org_ids = org_payloads_and_results_count.collect(&:id)
+
+    # Collect the results trend for each organization
+    results_trend = {}
+    org_ids.each do |org_id|
+      # Get the DB records corresponding to the org.
+      # In some cases, the number of results can be less than the number of weeks
+      org_results_trends = org_results_trends_count.select{ |org_results_trend| org_results_trend.id == org_id }
+
+      results_trend[org_id] = []
+
+      (0..weeks - 1).each do |week_idx|
+        week_date = (week_idx + 1).weeks.ago.at_beginning_of_week
+
+        # Try to find the data for the current week date
+        week_trend = org_results_trends.find{ |org_trend| org_trend.trend_date == week_date }
+
+        # Check if the DB returned data for the week date, otherwise consider as 0
+        results_trend[org_id] << (week_trend.nil? ? 0 : week_trend.test_results_count)
+      end
+    end
+
+    # Retrieve the count data for projects
+    organizations_projects_count = Organization
```

Comment:
Couldn't these project counts and the test counts below be retrieved along with the first query? It already has a join on the `projects` table, it would only need an additional join on `project_tests`.